### PR TITLE
[Redis] `az redis import/export`: Add `--storage-subscription-id` support while importing/exporting data

### PR DESCRIPTION
**Pull request by ajtiwari07**
**Related command**

**az redis**

**Description**
PR is to update azure cli management client with new api v2023-04-01. Following parameters are added:
- New `storage-subscription-id` parameter added while performing import data to the cache.
- New `storage-subscription-id` parameter added while performing exporting data from the cache.
Swagger changes can be found here : 
https://github.com/Azure/azure-rest-api-specs/pull/23726

**Testing Guide**
Once Managed Identity is enabled on the cache & appropriate access roles specified in storage account then below commands can be run to export/import data on a storage account in a different subscription: https://learn.microsoft.com/en-us/azure/azure-cache-for-redis/cache-managed-identity
`az redis export -n \'{name}\' -g \'{rg}\' --prefix \'{prefix}\' --container \'{containersasURL}\' --preferred-data-archive-auth-method ManagedIdentity --storage-subscription-id \'{storageSubscriptionId}\' : optional parameter to provide subscription id where the storage account resides. 
`az redis import -n \'{name}\' -g \'{rg}\' --files \'{filesasURL}\' --preferred-data-archive-auth-method ManagedIdentity --storage-subscription-id \'{storageSubscriptionId}\' : optional parameter to provide subscription id where the storage account resides. 

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [x] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
Thank you for your contribution! We will review the pull request and get back to you soon.

**Comment by ajtiwari07** [comment]
AzureCLI-FullTest

**Comment by yonzhan** [comment]
Please fix CI issues.

**Comment by ajtiwari07** [comment]
> Please fix CI issues.

All Tests are running locally for me. Can you please trigger them since, i'm not aware of git shortcut.

**Comment by evelyn-ys** [comment]
@ajtiwari07 Please sync with latest CLI dev branch and run `azdev setup -c` again before you re-record tests

**Comment by ajtiwari07** [comment]
@evelyn-ys I have updated my latest fork, re-ran all the live tests and they all ran successfully. Still I'm seeing pipeline failures, could advise on how to fix them?
image

**Comment by evelyn-ys** [comment]
@ajtiwari07 Please sync with latest dev branch and run `azdev setup -c` again before re-recording tests

**Comment by evelyn-ys** [comment]
Please resolve conflicts

File: src/azure-cli/azure/cli/command_modules/servicefabric/_params.py [hunk 1]
```diff
@@ -323,7 +323,7 @@ def load_arguments(self, _):  # pylint: disable=too-many-statements
         c.argument('vm_size', help='The size of virtual machines in the pool. All virtual machines in a pool are the same size.', default='Standard_D2')
         c.argument('vm_image_publisher', help='The publisher of the Azure Virtual Machines Marketplace image.', default='MicrosoftWindowsServer')
         c.argument('vm_image_offer', help='The offer type of the Azure Virtual Machines Marketplace image.', default='WindowsServer')
-        c.argument('vm_image_sku', help='The SKU of the Azure Virtual Machines Marketplace image.', default='2019-Datacenter')
+        c.argument('vm_image_sku', help='The SKU of the Azure Virtual Machines Marketplace image.', default='2022-Datacenter')
```
Comment (evelyn-ys) [inline src/azure-cli/azure/cli/command_modules/servicefabric/_params.py:326]: Why you change the parameter default value for servicefabric module? Can you make one PR clean just for one module?

Comment (ajtiwari07) [reply src/azure-cli/azure/cli/command_modules/servicefabric/_params.py]: Undid the modified defaults,

File: src/azure-cli/azure/cli/command_modules/redis/recoveryconfig_sql_restore.json [hunk 2]
```diff
@@ -0,0 +1 @@
+{"alternate_directory_paths":null,"container_id":null,"container_uri":"VMAppContainer;compute;sql-clitest-rg;sql-clitestvm-donotuse2","database_name":null,"filepath":null,"item_type":"SQL","item_uri":"SQLDataBase;mssqlserver;msdb","log_point_in_time":null,"recovery_mode":null,"recovery_point_id":"55886027005995","restore_mode":"OriginalLocation","source_resource_id":"/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/sql-clitest-rg/providers/Microsoft.Compute/virtualMachines/sql-clitestvm-donotuse2","workload_type":"SQLDataBase"}
```
[the remark this answers is not in the document]
Comment (ajtiwari07) [reply src/azure-cli/azure/cli/command_modules/redis/recoveryconfig_sql_restore.json]: Removed all recovery files.

**Review by evelyn-ys — APPROVED** [review]

**Review by kairu-ms — APPROVED** [review]

**Review by yanzhudd — APPROVED** [review]

**Pull request merged**